### PR TITLE
Create Issue and Pull Request templates

### PR DESCRIPTION
**Pull request by sgibson91**
Hey! I've created a couple of templates for issues and pull requests that hopefully will encourage contributions to be as detailed as possible so they can be reviewed/actioned as quickly as possible!

I've gone with just one "support" issue template rather than a more traditional bug report/feature request template since I don't think users really know if they're reporting a bug or asking for a feature when they open an issue. I think the interaction usually goes along the lines of:

User: Hey! I tried to do X but couldn't get it to work?
[Some discussion with Maintainer happens]
Maintainer: Yes, it appears this is a bug - we will release a patch ASAP
or
Maintainer: We don't support this use case.
[Decision process on whether use case is within scope of the project or not. If yes, scope a feature implementation.]

However, I'm very happy to have discussions on this 😄

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/support.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+name: Support Request
+about: Get some help using this repository
+labels: support
```
Comment (JimMadge) [inline .github/ISSUE_TEMPLATE/support.md:4]: I presume we have to create this label for this to work?

Comment (sgibson91) [reply .github/ISSUE_TEMPLATE/support.md:4]: Yes, but can be created after merge :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+-->
+
+### Summary
+<!--
+Describe the problem to be fixed or feature to be implemented in this Pull Request.
+Please reference any related issue(s) and use fixes/close keywords to automatically close them, if pertinent.
+For example: "fix #58", or "addresses #238"
+-->
+
+fix #<!-- NUM -->
```
Comment (JimMadge) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Describe the problem to be fixed or feature to be implemented in this Pull Request.
Please reference any related issue(s) and use fixes/closes keywords to automatically close them, if pertinent.
For example: "fixes #58", or "related to #238"
-->

fixes #<!-- NUM -->
```

**Review by JimMadge — CHANGES_REQUESTED** [review]
Thanks @sgibson91 this is excellent. I really like these templates.

I would be happy to have a feature request template too, but I get the impression that your experience suggests it isn't that useful. So, lets merge this and deal with a feature request template if it ever feels like one is needed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (JimMadge) [reply .github/PULL_REQUEST_TEMPLATE.md]: Just me being a bit pedantic.

**Review by JimMadge — APPROVED** [review]

**Comment by sgibson91** [comment]
> I would be happy to have a feature request template too, but I get the impression that your experience suggests it isn't that useful. So, lets merge this and deal with a feature request template if it ever feels like one is needed.

A feature request template could ask questions like "What other approaches have you tried?", "What skills would be required to implement this feature?" and "How long do you expect implementation to take?" which would really help with scoping the proposed change and what the PR might look like. But I would also suggest that the user base of this repo would be less likely to know those things. As you say, cross the bridge when we arrive at it and I'm always happy to draft something up for it.

**Pull request merged**